worker/worker.go

```diff
@@ -6,9 +6,39 @@ import (
 	"encoding/json"
 	"errors"
 	"log/slog"
+	"strconv"
 	"sync"
 )
 
+// StringBool allows JSON booleans to be unmarshalled as strings.
+type StringBool string
+
+// UnmarshalJSON converts JSON booleans to strings for StringBool.
+func (sb *StringBool) UnmarshalJSON(b []byte) error {
+	var boolVal bool
+	err := json.Unmarshal(b, &boolVal)
+	if err == nil {
+		*sb = StringBool(strconv.FormatBool(boolVal))
+		return nil
+	}
+
+	var strVal string
+	err = json.Unmarshal(b, &strVal)
+	if err == nil {
+		*sb = StringBool(strVal)
+	}
+
+	return err
+}
+
+// String returns the string representation of the StringBool.
+func (sb StringBool) String() string {
+	return string(sb)
+}
+
+// OptimizationFlags is a map of optimization flags to be passed to the pipeline.
+type OptimizationFlags map[string]StringBool
```

Comment:
A small nit - leave it up to you if you want to address.

IIUC the motivation for this struct is that we want to have map to store all env vars and since env var values are always strings we need to make sure if we unmarshal a JSON with a bool value that the bool value is converted into a string. If this is the case, then it seems that something like `map[string]EnvValue` might be more descriptive as a type name.

Reply:
Good point!